[fix #211] 接入ca代码

jdyuankai opened the pull request.

jdyuankai committed:
[fix #2211] 加入ca代码

wmmair commented on scripts/front_end/server/main.go:
```diff
@@ -47,6 +52,43 @@ func main() {
 
+func ca_verify(signedData []byte, targetData []byte){
+	rootCertPath = "./root.cert"
+	userCertPath = "./user.cert"
```
The path could be configured.

jdyuankai replied:
好的，现在文件名和目录已经作为常量配置了。

jdyuankai committed:
[fix #221] 完善ca代码及接口

jdyuankai committed:
[fix #211] CA方法测试完成，新增功能如下：
1. 增加CA证书开关，默认打开，如需关闭加入启动参数 -ca=false
2. AddNode，QueryNodes增加参数Sign，OriData，原有权限相关参数删除。测试完毕
3. 增加AddNode，QueryNodes单元测试方法。
4. QueryNodes接口测试完成。修改接口测试bug，将Sign做url解码处理。

jdyuankai commented:
接口测试方法（测试数据）：
// 启动go client. 
cd ~/StreamNet/script/front_end/server
go run man.go -host=http://127.0.0.1:14700 -ca=false

// post a queryNodes query
curl -s http://localhost:8000/QueryNodes -X POST -H "Content-Type: application/json" -d "{\"Attester\":\"10.0.0.1\",\"Attestee\":\"10.0.0.2\",\"Score\":\"1\",\"Time\":\"2213223190\",\"Nonce\":\"1\",\"Address\":\"1vofXj4Vf2cgJDQrbbN2Zc6gG9qmRmk96\",\"Sign\":\"%7b%22cert%22%3a%22LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNkakNDQWh1Z0F3SUJBZ0lVZkpnTElSQitnSXEvVEkwMFR6YWJrUktGTkJrd0NnWUlLb1pJemowRUF3SXcKY3pFTE1Ba0dBMVVFQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaApiaUJHY21GdVkybHpZMjh4R1RBWEJnTlZCQW9URUc5eVp6RXVaWGhoYlhCc1pTNWpiMjB4SERBYUJnTlZCQU1UCkUyTmhMbTl5WnpFdVpYaGhiWEJzWlM1amIyMHdIaGNOTWpBd016QTFNRGt3TlRBd1doY05NakV3TXpBMU1Ea3cKTlRBd1dqQlVNUXN3Q1FZRFZRUUdFd0pEVGpFUU1BNEdBMVVFQ0JNSFFtVnBhbWx1WnpFUU1BNEdBMVVFQnhNSApRbVZwYW1sdVp6RU9NQXdHQTFVRUNoTUZWSEpwWVhNeEVUQVBCZ05WQkFzVENHVnVaMmx1WldWeU1Ga3dFd1lICktvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUVCcHV2d0tpRy9GTmZUeW5QWVhOcVZ1VVE0WkNPQXdVTEpURzkKVkRDT0ZQeEU0WW5LMXhJMU9iSmI2K1pVV0MyRVVzNDgxTEJuRkpFTzZTYVp2QkJDV0tPQnF6Q0JxREFPQmdOVgpIUThCQWY4RUJBTUNCYUF3SFFZRFZSMGxCQll3RkFZSUt3WUJCUVVIQXdFR0NDc0dBUVVGQndNQ01Bd0dBMVVkCkV3RUIvd1FDTUFBd0hRWURWUjBPQkJZRUZQcFdxRkVPQXc4MVpQN1lLbHEvZVlyTWVYWlRNQ3NHQTFVZEl3UWsKTUNLQUlIS1dKRFVMclg3WTFpRjVzZkxtTzlwSEJSbkRhS0lnRGJtSUxrdWNlZWsxTUIwR0ExVWRFUVFXTUJTQwpFalE1TGpJek15NHhPVEV1TmpBNk9EZzRPREFLQmdncWhrak9QUVFEQWdOSkFEQkdBaUVBcU14bXd2VituQ05QCm9jVHdLRzFNR010Z09ZeWFBaERkd3JTam9hVEV3MVlDSVFDZ21OeUZsZ1Qra2M3bVVEelZWNEZOZnNDZ2JFZzcKV25SeWpSSDBFNlowb3c9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg%3d%3d%22%2c%22signature%22%3a%22MEYCIQCrI7Xe%2fD2ftFU8G3vI2RJu6KY5IGmuVG%2btyaxCpPnv1QIhAMhC2QRgvVYWJiY76s2QKYjelxwRPYJ49LCk5LRE3pjJ%22%7d\",\"OriData\":\"abc\"}"

jdyuankai committed:
Merge branch 'dev' into add_ca

jdyuankai committed:
[fix #211] CA方法测试完成，新增功能如下：
删掉过时的且会引起歧义的参数
发起请求时sign需要做url encoding

jdyuankai committed:
[fix #211] CA方法测试完成，新增功能如下：
打印生成的签名

jdyuankai committed:
[fix #211] CA方案，删除原有的验签方式

jdyuankai committed:
Merge branch 'add_ca' of https://github.com/triasteam/StreamNet into add_ca

jdyuankai committed:
Update .travis.yml

jdyuankai committed:
[fix #211] CA 接入说明

jdyuankai committed:
[fix #211] python 固定python包版本

jdyuankai committed:
Merge branch 'add_ca' of https://github.com/triasteam/StreamNet into add_ca

# Conflicts:
#	.travis.yml

jdyuankai committed:
[fix #211] print git account in travis ci

jdyuankai committed:
[fix #211] remove mock version limit

jdyuankai committed:
[fix #211]  checkout same version for test with i

jdyuankai committed:
[fix #211] fix travis

jdyuankai committed:
[fix #211] fix travis

jdyuankai committed:
[fix #211] debug travis

jdyuankai committed:
[fix #211] fix travis

jdyuankai committed:
[fix #211] fix travis

jdyuankai committed:
[fix #211] remote print
